extensions/desktop/gnome/launcher-specific

```diff
@@ -11,6 +11,7 @@ if [ "$wayland_available" = true ]; then
   export QT_QPA_PLATFORM=wayland-egl
 fi
 append_dir GTK_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/gtk-3.0"
+append_dir PYTHONPATH "$SNAP_DESKTOP_RUNTIME/usr/lib/python3.6/site-packages"
```

Comment:
I don't really like hardcoding `python3.6` here but it does work.  
What would be a better way of determining the python version though? On a system with the gnome3-34-1804 snap installed, I see several versions:
```
$ ll /snap/gnome-3-34-1804/current/usr/lib | grep python
drwxr-xr-x  27 root root     4327 Jun  1 07:55 python2.7
drwxr-xr-x   3 root root       36 Jan 29  2019 python3
drwxr-xr-x  30 root root     3628 Jun  1 07:54 python3.6
drwxr-xr-x   4 root root       47 Oct 28  2019 python3.7
drwxr-xr-x   4 root root       47 Oct 28  2019 python3.8
```
I'd like to replace this one append_dir line with something like this:
```
export PYTHON_NUMBERS=`python3 --version | cut -d ' ' -f2 | sed 's/..$//'`
export PYTHON_RUNTIME="python$PYTHON_NUMBERS"
append_dir PYTHONPATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$PYTHON_RUNTIME/site-packages"
```
But I'm not sure if maybe PYTHON_NUMBERS would be better determined from what exists in $SNAP_DESKTOP_RUNTIME/usr/lib.

Thoughts?
```

Reply:
I imagine `launcher-specific` is quite coupled to the contents of the gnome platform snap, so I think it's fine that it is hardcoded there.  Once  a newer version of gnome platform comes out that breaks those paths (e.g. for core20) then that'll have to be forked.

Reply:
You could perhaps allow a flag (e.g. GNOME_PLATFORM_PYTHON_PATH=...) to be passed to `make` and have it insert the correct path when building it too. That may prevent forking and allow the extension to define it in the part.

Reply:
I suspect this problem goes away if the rogue python directories there are cleaned up, this probably showed up from use of stage-packages and the use of dist-packages. I would just support the default python for the base, as soon as you diverge this becomes a problem. Which actually triggers another test in my mind, we need to test the extension using a python different that the python that is default for the base and the implications of setting this PYTHONPATH, for which, your script is probably doing the right thing but check for the directories existance before exporting :-)

Reply:
Awesome thanks!! Ok sounds like hardcoding is the way to go for now then :)
I tested this with pick-colour-picker and works just fine.